### PR TITLE
define ClipList.format (0) to make it more futureproof

### PR DESCRIPTION
**Pull request by anthrotype**
Fixes https://github.com/googlefonts/colr-gradients-spec/issues/346

### CODE REVIEW

**Review by drott — APPROVED** [review]
LGTM, but I'll let @PeterConstable have a look as well. Thanks.

**Comment by PeterConstable** [comment]
We haven't been consistent with whether struct format sequence start at 0 or 1. For Paint, we start at 1, but for ClipList and ClipBox, it's 0.

Elsewhere in the OT spec, when there are multiple formats for a struct, usually the sequence starts at 1. Cmap subtables are the exception, but they also are exceptional in general (0, 2, 4, 6, 8, 10, 12, 13, 14).

It's a small issue, but I think we should have ClipList format set to 1, and change the ClipBox formats from 0 and 1 to 1 and 2.

**Comment by anthrotype** [comment]
ok, I don't mind either 0 or 1. I changed them to start from 1, let me know if that's ok

**Comment by PeterConstable** [comment]
Optional: 

>ClipBox table format 1, static clip box:

Change to

>ClipBoxFormat1 table, static clip box:

That's a pattern used elsewhere, particularly for structs related to OT Layout and in STAT:
* BaseCoordFormat1
* CoverageFormat1
* ClassDefFormat1
* SequenceContextFormat1
* ChainedSequenceContextFormat1
* CaretValueFormat1
* SinglePosFormat1
* PairPosFormat1
* ...
* AxisValueFormat1

But there are a couple of exceptions: not for cmap subtables, or for ItemVariationStore (which has only one format), or CFF2 FD Select (three formats: 0, 3 and 4).

Perhaps do this for ClipBox but not at this time for ClipList?

**Review by rsheeter — APPROVED** [review]

**Comment by rsheeter** [comment]
LGTM, don't have strong feelings about starting at 0 or 1.

File: colr-gradients-spec.md [hunk 1]
```diff
@@ -558,7 +558,12 @@ typedef ArrayOf<BaseGlyphPaintRecord, uint32> BaseGlyphList;
 // Only layers accessed via PaintColrLayers (format 1) need be encoded here.
 typedef ArrayOf<Offset32<Paint>, uint32> LayerList;
 
-typedef ArrayOf<Clip, uint32> ClipList;
+struct ClipList
+{
+  uint8                 format;  // Set to 1.
+  uint32                numClips;
+  Clip                  clips[/*numClips*/];  // Clip records, sorted by startGlyphID
```
Comment (behdad) [inline colr-gradients-spec.md:565]: s/`startGlyphID`/`gidStart`/.

Comment (behdad) [reply colr-gradients-spec.md:565]: I also feel like `Clip` should be renamed to something like `ClipBoxRecord` at least.

Comment (anthrotype) [reply colr-gradients-spec.md:565]: A ClipList already contains an array of ClipRecords, which in turn reference a ClipBox table. Having both a ClipBoxRecord and a ClipBox table may be too confusing.

Changing startGlyphID to gidStart can certainly be done, but it's bordering bikeshedding and I have already cut a fonttools release 😅

Comment (behdad) [reply colr-gradients-spec.md:565]: > Changing startGlyphID to gidStart can certainly be done, but it's bordering on bikeshedding and I have already cut a fonttools release 😅

I wasn't asking for a change. I see `gidStart` in the actual datastructure, and `startGlyphID` in the comment.

Comment (behdad) [reply colr-gradients-spec.md:565]: > ClipList already contains an array of ClipRecords

I feel like we are talking about the same thing.

Comment (anthrotype) [reply colr-gradients-spec.md:565]: oh right, I see

**Pull request merged**